internal/transport/http_util.go

```diff
@@ -388,15 +388,35 @@ func toIOError(err error) error {
 	return ioError{error: err}
 }
 
+type parsedDataFrame struct {
+	http2.FrameHeader
+	data mem.Buffer
+}
+
+func (df *parsedDataFrame) StreamEnded() bool {
+	return df.FrameHeader.Flags.Has(http2.FlagDataEndStream)
+}
+
 type framer struct {
 	writer *bufWriter
 	fr     *http2.Framer
+	reader io.Reader
+	// Cached data frame to avoid heap allocations.
+	dataFrame parsedDataFrame
```

Comment:
Nit: Can you please change this to a trailing line comment (since this is the only field with a comment), and doing so would make this struct definition easier to read visually.